### PR TITLE
Feat/add state countdown

### PR DESCRIPTION
**Pull request by hojarasca**
Showing days left for the different states of the proposal:

image

I added tests for functions that run purely in the server. So I created a new vitest config file.

### CODE REVIEW

**Comment by timbrinded** [comment]
I think it looks nicer as:
image


WDYT @tomimor ?

File: webapp/app/.server/service/proposal-times.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return {
+      totalDays: totalDays,
+      currentDay: currentDay,
+    };
+  }
+
+  if (status === PROPOSAL_STATES.Waiting) {
+    const days3 = daysInSeconds(3);
+    const days7 = daysInSeconds(7);
+    const vetoPeriodDuration = guardiansExtendedLegalVeto ? days7 : days3;
```
Comment (timbrinded) [inline webapp/app/.server/service/proposal-times.ts:27]: Enshrine this as a const at top of file so if this changes or needs to be refactored it's more obvious
(whether it's 7 days or 3, or even if it needs to be read from somewhere)

File: webapp/app/.server/service/proposal-times.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const delta = nowInSeconds - (creationTimestamp + vetoPeriodDuration);
+    const currentDay = Math.ceil(delta / daysInSeconds(1));
+    return {
+      totalDays: 30,
```
Comment (timbrinded) [inline webapp/app/.server/service/proposal-times.ts:31]: where does 30 come from?

File: webapp/app/routes/app/proposals/$id/_route.tsx [hunk 3]
```diff
@@ -444,10 +452,15 @@ function StatusIndicator({
...
 
-function displayProposalState(state: PROPOSAL_STATES) {
+function displayProposalState(
+  state: PROPOSAL_STATES,
+  times: { totalDays: number; currentDay: number } | null
+) {
   let color: string;
   let label: string;
 
+  const timeData = times ? `(day ${times.currentDay} out of ${times.totalDays})` : "";
```
Comment (timbrinded) [inline webapp/app/routes/app/proposals/$id/_route.tsx:462]: Pending Tomi's opinion, consider changing this to `( Day <current> of <total> )`
for cleaner look

File: webapp/app/routes/app/proposals/$id/proposal-state.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { PROPOSAL_STATES, type StatusTime } from "@/utils/proposal-states";
+
+export type ProposalStateProps = {
+  status: PROPOSAL_STATES;
+  times: StatusTime | null;
+};
+
+export default function ProposalState({ status, times }: ProposalStateProps) {
```
Comment (timbrinded) [inline webapp/app/routes/app/proposals/$id/proposal-state.tsx:8]: Include comment with link to the [schedule](https://docs.zknation.io/zksync-governance/schedule-1-standard-governance-procedures) where presumably you got this info from

File: webapp/app/routes/app/proposals/$id/proposal-state.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    case PROPOSAL_STATES.Ready:
+      color = "text-green-400";
+      label = "READY";
+      break;
+    case PROPOSAL_STATES.Expired:
+      color = "text-red-400";
+      label = "EXPIRED";
+      break;
+    case PROPOSAL_STATES.Done:
+      color = "text-green-400";
```
Comment (timbrinded) [inline webapp/app/routes/app/proposals/$id/proposal-state.tsx:40]: `Ready` and `Done` share the same colour. Perhaps delineate them?

File: webapp/app/test/unit/proposal-state.test.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      />
+    );
+
+    const element = await screen.findByText(/LEGAL VETO PERIOD/);
+    expect(element).toBeInTheDocument();
+    expect(element).toHaveClass("text-yellow-400");
+    expect(element).toHaveTextContent(/^LEGAL VETO PERIOD \(day 6 out of 13\)$/);
+  });
+
+  // it("renders Navbar with Logo and correct link", async () => {
```
Comment (timbrinded) [inline webapp/app/test/unit/proposal-state.test.tsx:35]: can remove these comments v

File: webapp/package.json [hunk 7]
```diff
@@ -20,6 +20,7 @@
     "db:studio": "drizzle-kit studio",
     "db:up": "docker compose up -d db",
     "db:down": "docker compose down db",
+    "test:server": "vitest run --config ./vitest.server.config.ts",
```
Comment (timbrinded) [inline webapp/package.json:23]: if you continue the lazy trend, you could also add test:s as another script :)

File: webapp/app/routes/app/proposals/$id/proposal-state.tsx (hunk 4 above)
Comment (hojarasca) [reply webapp/app/routes/app/proposals/$id/proposal-state.tsx:8]: I actually took this from the contract, not the docs. But I feel that linking the docs is better. Adding this.

File: webapp/app/test/unit/proposal-state.test.tsx (hunk 6 above)
Comment (hojarasca) [reply webapp/app/test/unit/proposal-state.test.tsx:35]: Done.

File: webapp/app/routes/app/proposals/$id/proposal-state.tsx (hunk 5 above)
Comment (hojarasca) [reply webapp/app/routes/app/proposals/$id/proposal-state.tsx:40]: Legal veto and waiting are also the same color. I'm happy to change the color, but maybe we should think a better meaning for the colors then.

File: webapp/app/routes/app/proposals/$id/_route.tsx (hunk 3 above)
Comment (hojarasca) [reply webapp/app/routes/app/proposals/$id/_route.tsx]: I also belive it looks nicer. I applied this.

File: webapp/app/.server/service/proposal-times.ts (hunk 2 above)
Comment (hojarasca) [reply webapp/app/.server/service/proposal-times.ts]: I took this from the contract. In the docs is documented as `Risk Review Period`. I removed to it's own better named and explained constant.

**Review by timbrinded — APPROVED** [review]
lgtm

**Pull request merged**